src/test/java/com/scottlogic/sorts/KeywordPostSorterTest.java

```diff
@@ -0,0 +1,113 @@
+package com.scottlogic.sorts;
+
+import com.scottlogic.SortOrder;
+import com.scottlogic.UserPost;
+import com.scottlogic.filters.KeywordPostFilter;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class KeywordPostSorterTest {
+    UserPost keywords0 = new UserPost("jane smith",
+            OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC),
+            "Hello World!", 2);
+
+    UserPost keywords1 = new UserPost("joe bloggs",
+            OffsetDateTime.of(2020, 1, 3, 8, 53, 34, 0, ZoneOffset.UTC),
+            "Another example post", 1);
+
+    UserPost keywords2 = new UserPost("joe bloggs",
+            OffsetDateTime.of(2020, 3, 12, 13, 22, 12, 0, ZoneOffset.UTC),
+            "An example of a post \nwith lines breaks and post", 3);
+
+    UserPost keywords3 = new UserPost("joe bloggs",
+            OffsetDateTime.of(2020, 3, 12, 13, 22, 12, 0, ZoneOffset.UTC),
+            "A post example of a post \nwith lines breaks and post", 3);
+
+    UserPost keywords1only = new UserPost("Jane Smith",
+            OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC),
+            "Post", 2);
+
+    UserPost keywords1_2 = new UserPost("Joe Bloggs",
+            OffsetDateTime.of(2020, 1, 3, 8, 53, 34, 0, ZoneOffset.UTC),
+            "Another example post.", 1);
+
+    UserPost keywords0_2 = new UserPost("Joe Bloggs",
```

Comment:
Really nice naming of the variables here makes the future tests a lot easier to read and understand quickly